examples/coiled/terraclimate.ipynb

```diff
@@ -0,0 +1,249 @@
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import coiled\n",
+    "import numpy as np\n",
+    "import xarray as xr\n",
+    "\n",
+    "from virtualizarr import open_virtual_dataset\n",
+    "from virtualizarr.backend import FileType"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Create the Terraclimate variable and year url combinations \n",
+    "`14 variables * 66 years = 924 NetCDF files`\n",
+    "\n",
+    "\n",
+    "\n"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "tvars = [\n",
+    "    \"aet\",\n",
+    "    \"def\",\n",
+    "    \"pet\",\n",
+    "    \"ppt\",\n",
+    "    \"q\",\n",
+    "    \"soil\",\n",
+    "    \"srad\",\n",
+    "    \"swe\",\n",
+    "    \"tmax\",\n",
+    "    \"tmin\",\n",
+    "    \"vap\",\n",
+    "    \"ws\",\n",
+    "    \"vpd\",\n",
+    "    \"PDSI\",\n",
+    "]\n",
+    "min_year = 1958\n",
+    "max_year = 2023\n",
+    "time_list = np.arange(min_year, max_year + 1, 1)\n",
+    "\n",
+    "combinations = [\n",
+    "    f\"https://climate.northwestknowledge.net/TERRACLIMATE-DATA/TerraClimate_{var}_{year}.nc\"\n",
+    "    for year in time_list\n",
+    "    for var in tvars\n",
+    "]"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Define the coiled serverless function\n",
+    "\n",
+    "### Serverless function setup notes:\n",
+    "- This coiled function is tailored to AWS\n",
+    "- `vm_type=[\"t4g.small\"]` - This is a small instance, you shouldn't need large machines for reference generation\n",
+    "- `spot_policy=\"spot_with_fallback\"` is cheaper, but might have unintended consequences\n",
+    "- `arm=True` uses VMs with ARM architecture, which is cheaper\n",
+    "- `idle_timeout=\"10 minutes\"` workers will shut down after 10 minutes of inactivity \n",
+    "- `n_workers=[100, 300]` adaptive scaling between 100 & 300 workers\n",
+    "- `name` [optional] if you want to keep track of your cluster in the coiled dashboard\n",
+    "\n",
+    "More details can be found in the [serverless function API](https://docs.coiled.io/user_guide/functions.html#api)."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "@coiled.function(\n",
+    "    region=\"us-west-2\",\n",
+    "    vm_type=[\"t4g.small\"],\n",
+    "    spot_policy=\"spot_with_fallback\",\n",
+    "    arm=True,\n",
+    "    idle_timeout=\"10 minutes\",\n",
+    "    n_workers=[100, 300],\n",
+    "    name=\"parallel_reference_generation\",\n",
+    ")\n",
+    "def process(filename):\n",
+    "    vds = open_virtual_dataset(\n",
+    "        filename,\n",
+    "        decode_times=True,\n",
+    "        loadable_variables=[\"time\", \"lat\", \"lon\", \"crs\"],\n",
+    "        filetype=FileType(\"netcdf4\"),\n",
```

Comment:
I dont' think you need to import `FileType`, you should just be able to pass `filetype='netcdf4'`.